### PR TITLE
Showing objective reminder for traitors now show their uplink code

### LINKED ISSUE #7971: The antagonist objectives reminder should show your uplink code
Title.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #7971

CL: [Improvement] hitting the objective reminder action button now shows traitors and nuclear operatives their uplink codes.

### CODE REVIEW

**Review by corp-0 — DISMISSED** [review]

**Comment by MaxIsJoe** [comment]
ah why is tests failing?

**Comment by MaxIsJoe** [comment]
fixed it

**Comment by ThatDan123** [comment]
GetObjectivesForPlayer(), is used for more things. This will duplicate the code for when traitors start, and put the code in the webhook. Make a new function to return it and then just add it for the antag action button.

**Review by craftxbox — DISMISSED** [review]
this will help ease the stream of people AHELPing for pda codes

**Comment by MaxIsJoe** [comment]
Where's the antag reminder button? I'm unable to find the thing that generates it

**Comment by craftxbox** [comment]
> Where's the antag reminder button? I'm unable to find the thing that generates it

Bloody shoes up on your button panel

**Comment by MaxIsJoe** [comment]
> > Where's the antag reminder button? I'm unable to find the thing that generates it
> 
> Bloody shoes up on your button panel

I mean the code

**Comment by craftxbox** [comment]
I trace it as follows:
[UnityProject/Assets/ScriptableObjects/UIActionsGraphics/ShowObjectives.asset](https://github.com/unitystation/unitystation/blob/65c70368551995285d9efee661c9994fe5db3a6b/UnityProject/Assets/ScriptableObjects/UIActionsGraphics/ShowObjectives.asset) ->
https://github.com/unitystation/unitystation/blob/badf75ad861f560f3e1287adc58cd1cdc7098fcb/UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs#L967 ->
https://github.com/unitystation/unitystation/blob/2c8bc9dfb361624f0df5db68c0d25d77c5fc5e79/UnityProject/Assets/Scripts/Player/Mind.cs#L223 ->
https://github.com/unitystation/unitystation/blob/54bfee01c641d4d63c995c61652b2308d15813d9/UnityProject/Assets/Scripts/Systems/Antagonists/SpawnedAntag.cs#L68

**Comment by ThatDan123** [comment]
Add the logic in:

https://github.com/unitystation/unitystation/blob/2c8bc9dfb361624f0df5db68c0d25d77c5fc5e79/UnityProject/Assets/Scripts/Player/Mind.cs#L223

Just double check whats calling it.

**Comment by MaxIsJoe** [comment]
review pls

File: UnityProject/Assets/Scripts/Player/Mind.cs [hunk 1]
```diff
@@ -223,8 +224,20 @@ public bool IsOnline()
...
+        {
+        	var playerInventory = body.DynamicItemStorage.GetItemSlots();
+        	foreach (var item in playerInventory)
+        	{
+        		if(item.IsEmpty) continue;
+        		if (item.ItemObject.TryGetComponent<PDALogic>(out var PDA) == false) continue;
+        		if(PDA.IsUplinkCapable == false) continue;
+                Chat.AddExamineMsgFromServer(playerMob, antag.GetObjectivesForPlayer());
+	        }
+        }
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Player/Mind.cs]:
```suggestion
		var playerMob = GetCurrentMob();
		
		//Send Objectives
		Chat.AddExamineMsgFromServer(playerMob, antag.GetObjectivesForPlayer());
		
		if (playerMob.TryGetComponent<PlayerScript>(out var body) == false) return;
		if (antag.Antagonist.AntagJobType == JobType.TRAITOR || antag.Antagonist.AntagJobType == JobType.SYNDICATE)
        {
        	var playerInventory = body.DynamicItemStorage.GetItemSlots();
        	foreach (var item in playerInventory)
        	{
        		if (item.IsEmpty) continue;
        		if (item.ItemObject.TryGetComponent<PDALogic>(out var PDA) == false) continue;
        		if(PDA.IsUplinkCapable == false) continue;
        		
        		//Send Uplink code
                Chat.AddExamineMsgFromServer(playerMob, $"PDA uplink code retrieved: {PDA.UplinkUnlockCode}");
	        }
        }
```

**Comment by MaxIsJoe** [comment]
added suggestion

**Review by ThatDan123 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**